include/Board.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <memory>
+#include <map>
+#include <vector>
+#include <iostream>
+
+#include "PieceFactory.h"
+
+class Piece;
+
+class Board {
+public:
+    static Board& getInstance();
+    int move(const std::string&);
+    std::pair<std::pair<int, int>, std::pair<int, int>> parseMoveString(const std::string&) const;
+    bool isValidPosition(int, int) const;
+    void fillBoard(const std::string&);
+
+private:
+    bool m_currentPlayer = 1;
+    static constexpr int m_numRows = 8;
+    static constexpr int m_numCols = 8;
+
+    std::vector<std::vector<std::unique_ptr<Piece>>> m_chessMatrix;
+
+    // Functions used in move()
+    void makeMove(std::pair<std::pair<int, int>, std::pair<int, int>>&);
```

Comment:
The Board class seems to have multiple responsibilities, such as maintaining the game state and handling move operations. Consider separating these responsibilities into different classes or functions to adhere to the Single Responsibility Principle.

Reply:
I split the Board class into 2 new classes: GameState, which maintains the game state, and MoveHandler, Which handles the move operations.

Reply:
Excellent very good job